src/Altinn.Platform/Altinn.Platform.Profile/Profile/Services/Implementation/UserProfilesWrapper.cs

```diff
@@ -20,76 +22,94 @@ public class UserProfilesWrapper : IUserProfiles
     {
         private readonly ILogger _logger;
         private readonly GeneralSettings _generalSettings;
+        private readonly IMemoryCache _memoryCache;
         private readonly HttpClient _client;
+        private readonly MemoryCacheEntryOptions _cacheOptions;
+        private readonly JsonSerializerOptions _serializerOptions;
 
         /// <summary>
         /// Initializes a new instance of the <see cref="UserProfilesWrapper"/> class
         /// </summary>
         /// <param name="httpClient">HttpClient from default http client factory</param>
         /// <param name="logger">the logger</param>
         /// <param name="generalSettings">the general settings</param>
-        public UserProfilesWrapper(HttpClient httpClient, ILogger<UserProfilesWrapper> logger, IOptions<GeneralSettings> generalSettings)
+        /// <param name="memoryCache">the memory cache</param>
+        public UserProfilesWrapper(
```

Comment:
This client implements the IUserProfiles interface. Instead of changing the implementation of the client and adding caching to it's responsibilities you could create a decorator class and handle the caching there. That way we don't have to change an existing implementation, testing of caching gets easier and it's easier to change the cache implementation later if you would like to use a distributed cache for instance.